iceoryx_utils/include/iceoryx_utils/internal/cxx/variant_queue.inl

```diff
@@ -78,156 +83,184 @@ optional<ValueType> VariantQueue<ValueType, Capacity>::push(const ValueType& val
             m_fifo.template get_at_index<static_cast<uint64_t>(VariantQueueTypes::SoFi_SingleProducerSingleConsumer)>()
                 ->push(value, overriddenValue);
 
-        if (!hadSpace)
-        {
-            ret = cxx::make_optional<ValueType>(std::move(overriddenValue));
-        }
-        break;
+        return (hadSpace) ? cxx::nullopt : cxx::make_optional<ValueType>(overriddenValue);
     }
     case VariantQueueTypes::FiFo_MultiProducerSingleConsumer:
     {
         auto hadSpace =
             m_fifo.template get_at_index<static_cast<uint64_t>(VariantQueueTypes::FiFo_MultiProducerSingleConsumer)>()
                 ->tryPush(value);
 
-        if (!hadSpace)
-        {
-            ValueType droppedValue = value;
-            ret = cxx::make_optional<ValueType>(std::move(droppedValue));
-        }
-        break;
+        return (hadSpace) ? cxx::nullopt : cxx::make_optional<ValueType>(value);
     }
     case VariantQueueTypes::SoFi_MultiProducerSingleConsumer:
     {
-        auto overriddenValue =
-            m_fifo.template get_at_index<static_cast<uint64_t>(VariantQueueTypes::FiFo_MultiProducerSingleConsumer)>()
-                ->push(value);
-        if (overriddenValue)
-        {
-            ret = cxx::make_optional<ValueType>(std::move(overriddenValue.value()));
-        }
-        break;
+        return m_fifo
+            .template get_at_index<static_cast<uint64_t>(VariantQueueTypes::FiFo_MultiProducerSingleConsumer)>()
+            ->push(value);
+    }
+    case VariantQueueTypes::BlockingFiFo_SingleProducerSingleConsumer:
+    {
+        auto fifoShouldBeDestroyed = m_fifo
+                                         .template get_at_index<static_cast<uint64_t>(
+                                             VariantQueueTypes::BlockingFiFo_SingleProducerSingleConsumer)>()
+                                         ->push(value);
+
+        return (fifoShouldBeDestroyed) ? cxx::nullopt : cxx::make_optional<ValueType>(value);
```

Comment:
Should we report here that the underlying queue is unusable? Same for `BlockingFiFo_MultiProducerSingleConsumer`

Reply:
Nope since this can be valid behavior where we dont want a warning. Just think of a publisher which is waiting in here, then we press control c and clean up everything and we enter exactly this case. Then the user would always get a warning here or some other message. This must be handled on a higher level.

Btw. this would also be the case when we have a thread where the publisher is waiting and we would like to terminate the thread in a clean fashion. We would mark the queue as to be destroyed and the publisher in that thread should awake - without any message from here - this has to be handled there.